### PR TITLE
replace spaces by dashes when normalizing usernames

### PR DESCRIPTION
**Pull request by kamatsuoka**

### CODE REVIEW

**Comment by kamatsuoka** [comment]
This is for https://github.com/jupyterhub/the-littlest-jupyterhub/issues/832

File: tljh/normalize.py [hunk 1]
```diff
@@ -8,14 +8,15 @@ def generate_system_username(username):
...
+    If username < 26 char, we just return it
+    (with any spaces replaced by dashes).
     Else, we hash the username, truncate username at
     26 char, append a '-' and first add 5char of hash.
     This makes sure our usernames are always under 32char.
     """
 
     if len(username) < 26:
-        return username
+        return username.replace(' ', '-')
```
Comment (manics) [inline tljh/normalize.py:19]: This results in a clash between `a-b c` and `a b-c`

Comment (kamatsuoka) [reply tljh/normalize.py:19]: Good point. So maybe it's better to remove any illegal characters and then append a hash of the original name, as we do for long usernames ...

**Comment by kamatsuoka** [comment]
Closing this in favor of https://github.com/jupyterhub/the-littlest-jupyterhub/pull/834

**Pull request closed without merging**